### PR TITLE
update hbbft to master branch

### PR DESCRIPTION
**Pull request by Maan2003**
We updated fedimint/hbbft master branch, but never changed it here.

Co-authored-by: Justin Moon <mail@justinmoon.com>

### CODE REVIEW

**Comment by justinmoon** [comment]
Looks good to me.

For context, [this commit](https://github.com/poanetwork/hbbft/pull/423/files#diff-72ef38e24b12b374b786a06af2218758fc57ceed9f1907d2dbdceff087ff7827R12) removed the public and private keys from `NetworkInfo`. I'm not sure exactly what these keys are used for, though ..

The other major difference is that the `minimint` branch we're currently using turns of this `simd-accel` feature. The (deleted) comment in the link above mentions wanting to turn `simd-accel` back on once [this issue](https://github.com/darrenldl/reed-solomon-erasure/issues/28) gets fixed. The issue was closed without a fix, but I'm able to run with `simd-accel` feature on m1 mac due to [this fix](https://github.com/rust-rse/reed-solomon-erasure/pull/92). I also was able to compile all the targets metioned in the issue, plus built flutter app on ios and android. So I don't think this `simd-accel` feature is a problem AFAICT.

File: minimint/src/net/peers.rs [hunk 1]
```diff
@@ -234,16 +234,20 @@ where
     async fn send(&mut self, target: Target<PeerId>, msg: T) {
         trace!(?target, "Sending message to");
         match target {
-            Target::All => {
-                for connection in self.connections.values_mut() {
-                    connection.send(msg.clone()).await;
+            Target::AllExcept(not_to) => {
```
Comment (elsirion) [inline minimint/src/net/peers.rs:237]: This will conflict with #299, see my comment there.

**Comment by elsirion** [comment]
I looked into https://github.com/poanetwork/hbbft/pull/423 and it is fascinating! I didn't realize HBBFT proper didn't even need non-threshold keys, it was only used to vote on joining peers (for the dynamic variant of HBBFT).

A second nice change is that apparently so far HBBFT was committing to the entire network info each epoch and they switched to only committing to the validator set (peer set in our terminology). This sounds as if it would make HBBFT faster (especially for epochs with few items) since the size of data to commit to has a large influence on performance and the network info is quite a bit of data that would be committed to every epoch.

So I think, under the assumption that @afck et al. were careful enough, switching to master seems like a good move. We might also want to check what POA is using in production. At some point either I have to get acquainted with HBBFT again to review the impl we are now using or we find someone else. @Kodylow maybe that's something for you?

**Review by elsirion — APPROVED** [review]
Code looks good, would like to get #299 in first though with the proposed changes there. Then this can be rebased (need it anyway due to `Cargo.lock`) and merged.

**Comment by jkitman** [comment]
@elsirion Does anyone know why we are on a `threshold_crypto` revision in the hbbft fork?
https://github.com/fedimint/hbbft/blob/master/Cargo.toml#L39

Unifying the `threshold_crypto` deps will allow us to generate our hbbft and ln-module keys using the same code.

**Comment by elsirion** [comment]
> @elsirion Does anyone know why we are on a `threshold_crypto` revision in the hbbft fork?
> https://github.com/fedimint/hbbft/blob/master/Cargo.toml#L39

Afaik there was an outdated dependency and the upstream project is abandoned (just as `hbbft`).

**Pull request merged**